rfcs/XXXX-gene-sets/text.md

```diff
@@ -0,0 +1,97 @@
+
+## Glossary
+
+- Gene set - List of genes
+- Differential expression - A statistic (typically logfold and p-value) describing the difference in gene expression between two sets of cells. It may refer to stats specific to one gene (differential expression value) or many (differential expression analysis). When referring to many genes it is typically capped (eg top 10 most differentially expressed genes)
+- LogFold - A ratio of the difference in expression levels. Typically calculated by taking the log2 of the gene expression value for each cell and then finding the mean of that value for each set of cells. Because the data is already log2 transformed you can then take the difference between the two means as the logfold value
+
+## Problem Statement | Background
+
+Gene sets are the expected output of differential expression analysis in cellxgene. Two (non-overlaping) sets of cells
+are selected and the difference in the expression level for each gene is computed by comparing the average
+expression level for each set of cells. When there is a particularly large difference in expression between the two sets of
+cells, that gene is identified and presented to the user. To ensure consistency and compatibility it is necessary to define a
+data format for storing references to these genes and any accompanying statistical data.
+
+Product Requirements
+
+### Users should be able to export differential expression results
+
+Currently cellxgene users can choose two sets of cells and compute/display the log fold and adjusted p-value of the top ten differentially expressed genes. Additionally we display a histogram of the gene expression levels of the selected cells for each gene in the right side bar.
+
+- [US1] A user should be able to export a gene set, if the gene set was created by differential expression, the pvalue, log fold change, and category label fields will be populated in addition to the gene name. This will be available in a csv file (see data formats below for more detail).
+
+### Users should be able to upload their own gene sets and see them in the right side bar with the option to apply them within current cellxgene functionalities, if the same gene ontology was used when generating the current dataset and the geneset uploaded by the user (ie the gene names match)
+
+- [US2] A user should be able to upload a geneset via a csv file (see data formats below for more detail) for a dataset.
+- [US3] A user should be able to upload a geneset they previously downloaded from cellxgene
+- [US4] When a user uploads a geneset, that list of genes should appear in the right side bar
+- [US5] If the user was logged in when they uploaded a geneset that genesit should persist if they logout and back in or reload the page
+- [US6] If the gene names match those used in the dataset the user should be able to use general cellxgene functionality (color by genes, calculate differential expression levels, etc.) on those genes
+
```

Comment:
Maybe this is outside the scope of this document, but what happens if a geneset contains genes that are not in the dataset?  Is that an error?

Reply:
By "calculate differential expression levels on those genes", do you mean that only those genes will be considered by the differential expression alg?    We can almost do that in the server, but probably some changes will need to be made.

Reply:
@bmccandless Thats a good question, I've been assuming If a geneset contains genes that arent in the dataset we will just assume their expression level is zero for the data set. But @signechambers1 may have thoughts.

Re  calculate differential expression levels on those genes" -- dont we already calculate diff exp for all genes in the dataset when comparing two sets of cells? Otherwise how would we determine what the top ten are? If the gene isnt in the dataset wouldnt the difference in expression between two sets of cells in the dataset just be zero?

Reply:
If a gene in a gene set isn't in the underlying data set we'd ideally like to notify the user and exclude the gene from the calculation of the mean expression of the gene set (we do not want to call the expression 0 cause that might not be true and artificially lower the mean). @colinmegill fyi if you have additional thoughts here

Reply:
I'm not sure this makes sense. Genes are outputs of diffexp, never inputs.

Reply:
@colinmegill I'm thinking of a situation where a user uploads a geneset from a paper (or elsewhere) and wants to calculate the diff exp values comparing cell sets within a cellxgene dataset for the genes in that geneset